VerilogCodingStyle.md

```diff
@@ -2406,6 +2518,45 @@ always_ff @(posedge clk) begin
 end
 ```
 
+### Assertion Macros
+
+It is encouraged to use SystemVerilog assertions (SVAs) throughout the design to
+check functional correctness and flag invalid conditions. In order to increase
+productivity and keep the assertions short and concise, the following assertion
+macros can be used:
+
+```systemverilog
+// immediate assertion, to be placed within a process.
+`ASSERT_I(<name>, <property>)
+// immediate assertion wrapped within an initial block. can be used for things
+// like parameter checking.
+`ASSERT_INIT(<name>, <property>)
+// concurrent assertion to be used for functional assertions.
+`ASSERT(<name>, <property>, <clk>, <reset condition>)
+// concurrent assertion that checks that a signal has a known value after reset
+// (i.e. that the signal is not `X`).
+`ASSERT_KNOWN(<name>, <signal>, <clk>, <reset condition>)
+```
+
+An implementation of these macros (including other useful variations thereof)
+can be found here:
+https://github.com/lowRISC/opentitan/blob/master/hw/ip/prim/rtl/prim_assert.sv
+
+
+#### A Note on Security Critical Applications
```

Comment:
Personally I think we can skip this section until we have full guidance here. But let's go ahead and leave it here, but please add something like "More security assertion and coding style guidance will be coming" else this seems like an odd incomplete story on the topic.

Reply:
Ok I added the following sentence at the end of this subsection:
```
More security assertion and coding style guidance will be given in a separate
document, soon.
```